driver/main.cpp

```diff
@@ -32,37 +62,63 @@ int wmain(int argc, wchar_t* argv[])
 int main(int argc, char* argv[])
 #endif
 {
-  if (argc < 3 || (!strcasecmp(argv[1], _SYS_STR("makegcn")) && argc < 3) ||
-      (!strcasecmp(argv[1], _SYS_STR("makewii")) && argc < 3) ||
-      (!strcasecmp(argv[1], _SYS_STR("mergegcn")) && argc < 4) ||
-      (!strcasecmp(argv[1], _SYS_STR("mergewii")) && argc < 4)) {
-    printHelp();
-    return 1;
-  }
-
   /* Enable logging to console */
   logvisor::RegisterStandardExceptions();
   logvisor::RegisterConsoleLogger();
+#if _WIN32
+  _setmode(_fileno(stdin), _O_U16TEXT);
+  _setmode(_fileno(stdout), _O_U16TEXT);
+  _setmode(_fileno(stderr), _O_U16TEXT);
+#endif
 
+  int argidx = 1;
+  nod::SystemString errand;
   bool verbose = false;
-  nod::ExtractionContext ctx = {true, [&](std::string_view str, float c) {
+  nod::Codepage_t disc_locale = CP_US_ASCII;
+  nod::ExtractionContext ctx = {true, [&](nod::SystemStringView str, float c) {
                                   if (verbose)
-                                    fmt::print(stderr, FMT_STRING("Current node: {}, Extraction {:g}% Complete\n"), str,
-                                               c * 100.f);
+                                    fmt::print(stderr, FMT_STRING(_SYS_STR("Current node: {}, Extraction {:g}% Complete\n")),
+                                               str, c * 100.f);
                                 }};
-  const nod::SystemChar* inDir = nullptr;
-  const nod::SystemChar* outDir = _SYS_STR(".");
-
-  for (int a = 2; a < argc; ++a) {
-    if (argv[a][0] == '-' && argv[a][1] == 'f')
+  while ( argidx < argc ) {
+    if (!strcasecmp(argv[argidx], _SYS_STR("-f"))) {
       ctx.force = true;
-    else if (argv[a][0] == '-' && argv[a][1] == 'v')
+      argidx++;
+      continue;
+    }
+    else if (!strcasecmp(argv[argidx], _SYS_STR("-v"))) {
       verbose = true;
+      argidx++;
+      continue;
+    }
+    else if (!strcasecmp(argv[argidx], _SYS_STR("-c"))) {
+      if (argidx+1 < argc ) {
+#if _WIN32
+        disc_locale = nod::StrToUL(argv[argidx+1], NULL, 0);
+#else
+        disc_locale = argv[argidx+1];
+#endif
+      }
+      else {
+        printHelp();
+        return 1;
+      }
+      argidx += 2;
+      continue;
+    }
+    else if (errand.empty()) {
+      errand = argv[argidx];
+      argidx++;
```

Comment:
Looks like you use a lot of postfix increment, please normalize to prefix increment for non-assignments.